riptide-logbook/src/main/java/org/zalando/riptide/logbook/LogbookPlugin.java

```diff
@@ -0,0 +1,48 @@
+package org.zalando.riptide.logbook;
+
+import lombok.AllArgsConstructor;
+import org.apiguardian.api.API;
+import org.zalando.logbook.Logbook;
+import org.zalando.logbook.Logbook.RequestWritingStage;
+import org.zalando.logbook.Logbook.ResponseProcessingStage;
+import org.zalando.riptide.Plugin;
+import org.zalando.riptide.RequestArguments;
+import org.zalando.riptide.RequestExecution;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+import static org.zalando.fauxpas.FauxPas.throwingConsumer;
+
+@API(status = EXPERIMENTAL)
+@AllArgsConstructor
+public final class LogbookPlugin implements Plugin {
+
+    private final Logbook logbook;
+
+    @Override
+    public RequestExecution aroundNetwork(final RequestExecution execution) {
+        return original -> {
+            // TODO is there a better way?!
+            final AtomicReference<ResponseProcessingStage> responseStage = new AtomicReference<>();
+
+            final RequestArguments arguments = original.withEntity(message -> {
+                final LocalRequest request = new LocalRequest(original);
+                final RequestWritingStage stage = logbook.process(request);
+                request.writeTo(message);
+                request.writeTo(message); // TODO oh no!
```

Comment:
Let's save CPU cycles here :wink: better test the state transitions in a unit test?

Reply:
the state should be transitioned into a state that doesn't allow further calls to `writeTo` even?